agent/http.go

```diff
@@ -210,7 +210,15 @@ func (s *HTTPHandlers) handler(enableDebug bool) http.Handler {
 	// handlePProf takes the given pattern and pprof handler
 	// and wraps it to add authorization and metrics
 	handlePProf := func(pattern string, handler http.HandlerFunc) {
+
 		wrapper := func(resp http.ResponseWriter, req *http.Request) {
+
+			// If enableDebug or ACL enabled, register wrapped pprof handlers
+			if !s.agent.enableDebug.Load() && s.checkACLDisabled() {
+				resp.WriteHeader(http.StatusNotFound)
+				return
+			}
```

Comment:
I think this should go after the ACL checks so if the ACL token is invalid we always return 403.

Reply:
please clarify - you mean `if !s.agent.enableDebug.Load() && s.checkACLDisabled() {` I should break this two flag in two ifs and return 403 if s.checkACLDisabled()? 
That will be breaking change right?

Reply:
@dhiaayachi

Reply:
nevermind, looking closer this is fine 😅